crates/api/src/chronicle_graphql/authorization.rs

```diff
@@ -0,0 +1,71 @@
+use base64::Engine;
+use jwtk::jwk::RemoteJwksVerifier;
+use serde_json::Value;
+use std::time::Duration;
+use thiserror::Error;
+use tracing::instrument;
+use url::Url;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Base64 decoding failure: {0}", source)]
+    Base64 {
+        #[from]
+        source: base64::DecodeError,
+    },
+    #[error("JSON decoding failure: {0}", source)]
+    Json {
+        #[from]
+        source: serde_json::Error,
+    },
+    #[error("JWT validation failure: {0}", source)]
+    Jwks {
+        #[from]
+        source: jwtk::Error,
+    },
+    #[error("application error: {0}", message)]
+    Simple { message: String },
+}
+
+pub struct JwtChecker {
+    verifier: RemoteJwksVerifier,
+}
+
+impl JwtChecker {
+    pub fn new(jwks_uri: &Url) -> Self {
+        Self {
+            verifier: RemoteJwksVerifier::new(jwks_uri.to_string(), None, Duration::from_secs(100)),
+        }
+    }
+
+    #[instrument(skip(self))]
```

Comment:
adding ret(Debug) here should eliminate the need to explictly debug!() it after, I also make this mistake a lot